File: prog3/trab1/src/Prog3_Trab1/TvSeriesSeason.java
package Prog3_Trab1;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.ListIterator;

final class TvSeriesSeason extends Media implements Serializable {

	private static final long serialVersionUID = 1L;
	protected static final int mediaTypeId = 1;
	private List<Person> mainActors = new ArrayList<Person>();
	private String seriesName;
	private int seasonLength;
	private int seasonNumber;
	
	public TvSeriesSeason(long mediaId,String name,String seriesName,int seasonLength,int seasonNumber,Genre genre) {
		super(mediaId,TvSeriesSeason.mediaTypeId,name,genre);
		this.seasonLength = seasonLength;
		this.seasonNumber = seasonNumber;
		this.seriesName = seriesName;
	}

	public final ListIterator<Person> getMainActorsIterator() {
		return this.mainActors.listIterator();
	}
	
	public final String getSeriesName() {
		return this.seriesName;
	}

	public final int getSeasonLength() {
		return this.seasonLength;
	}

	public final int getSeasonNumber() {
		return this.seasonNumber;
	}

	public final void setSeasonLength(int seasonLength) {
		this.seasonLength = seasonLength;
	}
	
	public final void setSeriesName(String seriesName) {
		this.seriesName = seriesName;
	}

	public final void setSeasonNumber(int seasonNumber) {
		this.seasonNumber = seasonNumber;
	}
	
	public final void addMainActors(Person actor) {
		this.mainActors.add(actor);
	}
	
	public static boolean isTvSeriesSeason(Media media) {
		return media.getMediaTypeId() == TvSeriesSeason.mediaTypeId;
	}
}

File: prog3/trab2/TvSeriesSeason.cpp
/**
 * TvSeriesSeason class implementation
 */

#include "TvSeriesSeason.hpp"

// Static values
int TvSeriesSeason::mediaTypeId = 1;

// Constructor
TvSeriesSeason::TvSeriesSeason(const long id, const string& name, const string& seriesName,
    const int seasonLengh, const int seasonNumber, Genre& genre)
    : Media(id, mediaTypeId, name, genre), seriesName(seriesName)
    , seasonLength(seasonLengh), seasonNumber(seasonNumber)
    {}

// Functions
void TvSeriesSeason::addActor(Person& person) {
    this->actors.push_back(&person);
}
bool TvSeriesSeason::isTvSeriesSeason(Media& media) {
    return (media.getTypeId() == mediaTypeId);
}

// Getters
string TvSeriesSeason::getSeriesName() {
    return seriesName;
}
int TvSeriesSeason::getSeasonLength() {
    return seasonLength;
}
int TvSeriesSeason::getSeasonNumber() {
    return seasonNumber;
}
list<Person*> TvSeriesSeason::getActors() {
    return actors;
}

// Setters
void TvSeriesSeason::setSeriesName(const string& seriesName) {
    this->seriesName = seriesName;
}
void TvSeriesSeason::setSeasonNumber(const int num) {
    this->seasonNumber = num;
}
void TvSeriesSeason::setSeasonLength(const int length) {
    this->seasonLength = length;
}

// Destructor
TvSeriesSeason::~TvSeriesSeason() {}
